base/linalg/bitarray.jl

```diff
@@ -211,3 +211,92 @@ function findmin(a::BitArray)
     k == l || return (false, ti)
     return m, mi
 end
+
+## transpose ##
+
+transpose(B::BitVector) = reshape(copy(B), 1, length(B))
+
+# fast 8x8 bit transpose from Henry S. Warrens's "Hacker's Delight"
+# http://www.hackersdelight.org/hdcodetxt/transpose8.c.txt
+function transpose8x8(x::UInt64)
+    y = x
+    t = xor(y, y >>> 7) & 0x00aa00aa00aa00aa
+    y = xor(y, t, t << 7)
+    t = xor(y, y >>> 14) & 0x0000cccc0000cccc
+    y = xor(y, t, t << 14)
+    t = xor(y, y >>> 28) & 0x00000000f0f0f0f0
+    return xor(y, t, t << 28)
+end
+
+function form_8x8_chunk(Bc::Vector{UInt64}, i1::Int, i2::Int, m::Int, cgap::Int, cinc::Int, nc::Int, msk8::UInt64)
+    x = UInt64(0)
+
+    k, l = get_chunks_id(i1 + (i2 - 1) * m)
```

Comment:
Alright, I gave in and fixed the intermediate commit. It is a bit nice to have that in the history.